lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -703,9 +703,17 @@ namespace Js
         bool isAnonymousFunction = function->IsAnonymousFunction();
 
         JavascriptLibrary* javascriptLibrary = function->GetType()->GetLibrary();
-        typeHandler->ConvertFunction(function, isAnonymousFunction ? javascriptLibrary->anonymousFunctionWithPrototypeTypeHandler : javascriptLibrary->functionWithPrototypeTypeHandler);
+        typeHandler->ConvertFunction(function, isAnonymousFunction ? javascriptLibrary->anonymousFunctionWithPrototypeTypeHandler : javascriptLibrary->functionWithPrototypeAndLengthTypeHandler);
         function->SetPropertyWithAttributes(PropertyIds::prototype, javascriptLibrary->CreateGeneratorConstructorPrototypeObject(), PropertyWritable, nullptr);
 
+        Var varLength;
+        GeneratorVirtualScriptFunction* scriptFunction = function->GetGeneratorVirtualScriptFunction();
+        if (!scriptFunction->GetProperty(scriptFunction, PropertyIds::length, &varLength, nullptr, scriptFunction->GetScriptContext()))
```

Comment:
Perhaps this is obvious elsewhere, but why would some generator functions not have lengths?

Reply:
(Same goes for async functions below)

Reply:
I'm not aware of any such case and the basic CI test suite passes (with a test-build on macos) with an AssertOrFailFast(false); added inside that if.

I took the logic of using 0 if the length can't be found from the GetPropertyBuiltIns method that I deleted from JavascriptGeneratorFunction.cpp - in general as my understanding of CC is far from complete I'm always nervous to remove any checks - if somehow it didn't have a length and I didn't have this check we would end up with a  nullptr de-reference.

Reply:
Id be interested to see if this could become an assert(orfailfast?) and if nothing in chakracore, chakrafull, test262, any other test collateral, I think its probably dead code. @boingoing mentioned something in this area this morning, though -- do you know the rationale here?

Reply:
I was waiting on hearing further comment on this point before updating.

Reply:
Makes sense. I think `AssertMsg(false, "why dont we have a length")` with a comment saying that the goal should be to remove this corner case if the Assert never hits is the way I would go forward.

Reply:
Added the assert and the comment.